core/src/datayoga_core/blocks/kafka/read/block.py

```diff
@@ -0,0 +1,92 @@
+import json
+import logging
+from abc import ABCMeta
+from itertools import count
+from typing import AsyncGenerator, List, Optional
+
+import orjson
+from confluent_kafka import Consumer, KafkaError, TopicPartition
+from datayoga_core import utils
+from datayoga_core.context import Context
+from datayoga_core.producer import Message
+from datayoga_core.producer import Producer as DyProducer
+
+logger = logging.getLogger("dy")
+
+
+class Block(DyProducer, metaclass=ABCMeta):
+    bootstrap_servers: str
+    group: str
+    topic: str
+    seek_to_beginning: bool
+    snapshot: bool
+    INTERNAL_FIELD_PREFIX = "__$$"
+    MSG_ID_FIELD = f"{INTERNAL_FIELD_PREFIX}msg_id"
+    MIN_COMMIT_COUNT = 10
+
+    def init(self, context: Optional[Context] = None):
+        logger.debug(f"Initializing {self.get_block_name()}")
+        connection_details = utils.get_connection_details(
+            self.properties["bootstrap_servers"], context)
+        logger.debug(f"Connection details: {json.dumps(connection_details)}")
```

Comment:
I suggest to move this debug log to the `get_connection_details` itself, so we will have it printed there whenever it's called.